src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,9 +1,79 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
+
+    public Object getInstance(Class<?> interfaceClazz) {
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (clazz.getInterfaces().length > 0
+                && !clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Can't create instance of a " + clazz.getSimpleName()
```

Comment:
this message is informative. if you're willing to keep it like this, you should check for `Component` annotation inside `findImplementation()`. Personally I would thought that we can't create instance because there is no annotation, but it's also possible when there no implementations of given interface